binary_search_tree/tree.py

```diff
@@ -14,46 +14,136 @@ class Tree:
+        current = self.root
+        while current:
+            if key == current.key:
+                return current.value
+            elif key <= current.key:
+                current = current.left
+            elif key > current.key:
+                current = current.right
+
+        return None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def inorder_helper(self, current, nodes = None):
+        if nodes is None:
+            nodes = []
+        if current:
+            self.inorder_helper(current.left, nodes)
+            nodes.append({
+                "key": current.key,
+                "value": current.value
+            })
+            self.inorder_helper(current.right, nodes)
+        return nodes
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def inorder(self):
-        pass
+        return self.inorder_helper(self.root)
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def preorder_helper(self, current, nodes = None):
+        if nodes is None:
+            nodes = []
+        if current:
+            nodes.append({
+                "key": current.key,
+                "value": current.value
+            })
+            self.preorder_helper(current.left, nodes)
+            self.preorder_helper(current.right, nodes)
+        return nodes
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def preorder(self):
-        pass
+        return self.preorder_helper(self.root)
 
-    # Time Complexity: 
-    # Space Complexity:     
+    def postorder_helper(self, current, nodes = None):
+        if nodes is None:
+            nodes = []
+        if current:
+            self.postorder_helper(current.left, nodes)
+            self.preorder_helper(current.right, nodes)
+            nodes.append({
+                "key": current.key,
+                "value": current.value
+            })
+        return nodes
+
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def postorder(self):
-        pass
+        return self.postorder_helper(self.root)
 
-    # Time Complexity: 
-    # Space Complexity:     
-    def height(self):
-        pass
+    def height_helper(self, current, nodes = None):
+        if nodes is None:
+            nodes = []
+        if current:
+            self.height_helper(current.left, nodes)
+            if current.left is None and current.right is None:
+                nodes.append(current.key)
+            self.height_helper(current.right, nodes)
+        return nodes
 
+    def get_count(self, key):
+        current = self.root
+        count = 0
+        while current:
+            count += 1
+            if key == current.key:
+                return count
+            elif key > current.key:
+                current = current.right
+            elif key < current.key:
+                current = current.left
+        return None
+
+    # Time Complexity: O(nlog(n))
+    # Space Complexity: 0(m) -> m == # of nodes
+    def height(self):
```

Comment:
✨ This time and space complexity is correct for your solution!

You could refactor your code to achieve an O(n) time and space solution. Instead of having height_helper return a list, have it return an integer. When you make a recursive call on the left/right subtree, you know that your height is 1 + the height of the deepest subtree. So you can say that the height is 1 + max(height of left subtree, height of right subtree).